src/DotVVM.Framework/ViewModel/Validation/ValidationErrorFactory.cs

```diff
@@ -91,7 +91,18 @@ public static ValidationResult CreateValidationResult<T>(ValidationContext valid
                 var expression = ExpressionUtils.Replace(e.expression, BindingExpressionBuilder.GetParameters(dataContext).First(p => p.Name == "_this"));
                 var jsast = translator.CompileToJavascript(expression, dataContext);
                 var pcode = BindingPropertyResolvers.FormatJavascript(jsast, niceMode: isDebug, nullChecks: false);
-                return JavascriptTranslator.FormatKnockoutScript(pcode, allowDataGlobal: true);
+                var script = JavascriptTranslator.FormatKnockoutScript(pcode, allowDataGlobal: true);
+
+                // Remove parentheses '(', ')' and right brackets ']'
+                script = System.Text.RegularExpressions.Regex.Replace(script, @"\(|\)|\]", "");
+                // Replace dots '.' and left brackets '[' with slash '/'
+                script = System.Text.RegularExpressions.Regex.Replace(script, @"\.|\[", "/");
+
+                // $rawData represents root of viewModel
+                if (script == "$rawData")
+                    script = "/";
+
+                return script;
```

Comment:
This seems very hacky - I think we should rather implement a simple visior specific for this purpose - it will clearly only support member access and indexers. This one would fail quite randomly would the user supply method invocations, conditions, ....